mono/metadata/locales.c

```diff
@@ -530,22 +532,40 @@ get_current_locale_name (void)
 	return ret;
 }
 
-MonoString*
-ves_icall_System_Globalization_CultureInfo_get_current_locale_name (void)
+static void
+culture_info_get_current_locale_name (MonoStringHandle ret, MonoError *error)
 {
 	gchar *locale;
-	MonoString* ret;
 	MonoDomain *domain;
 
+	mono_error_init (error);
+
 	locale = get_current_locale_name ();
-	if (locale == NULL)
-		return NULL;
+	if (locale == NULL) {
+		MONO_HANDLE_ASSIGN (ret, NULL);
+		return;
+	}
 
 	domain = mono_domain_get ();
-	ret = mono_string_new (domain, locale);
+	MONO_PREPARE_GC_CRITICAL_REGION;
```

Comment:
I agree, we should either make `mono_string_new_checked` return a MonoHandle and migrate user references to it, or create a new `mono_string_new_handle` (or another name)